app/services/encryption/kms_client.rb

```diff
@@ -27,16 +27,48 @@ def use_kms?(ciphertext)
     end
 
     def encrypt_kms(plaintext)
-      ciphertext_blob = aws_client.encrypt(
+      if plaintext.bytesize > 4096
+        encrypt_in_chunks(plaintext)
+      else
+        KEY_TYPE[:KMS] + encrypt_raw_kms(plaintext)
+      end
+    end
+
+    # chunk plaintext into ~4096 byte chunks, but not less than 1024 bytes in a chunk if chunking.
+    # we do this by counting how many chunks we have and adding one.
+    # :reek:FeatureEnvy
+    def encrypt_in_chunks(plaintext)
+      plain_size = plaintext.bytesize
+      number_chunks = plain_size / 4096
+      chunk_size = plain_size / (1 + number_chunks)
+      ciphertext = plaintext.scan(/.{1,#{chunk_size}}/).map(&method(:encrypt_raw_kms)).to_json
```

Comment:
We could also base64 encode the raw KMS result here, but I don't think that is a must have.